gradle-palantir-java-format/src/main/java/com/palantir/javaformat/gradle/PalantirJavaFormatProviderPlugin.java

```diff
@@ -18,36 +18,87 @@
         Preconditions.checkState(
                 rootProject == rootProject.getRootProject(),
                 "May only apply com.palantir.java-format-provider to the root project");
 
-        Configuration configuration = rootProject.getConfigurations().create(CONFIGURATION_NAME, conf -> {
-            conf.setDescription("Internal configuration for resolving the palantir-java-format implementation");
-            conf.setVisible(false);
-            conf.setCanBeConsumed(false);
-
-            conf.defaultDependencies(deps -> {
-                deps.add(rootProject
-                        .getDependencies()
-                        .create(ImmutableMap.of(
-                                "group", "com.palantir.javaformat",
-                                "name", "palantir-java-format",
-                                "version",
-                                        JavaFormatExtension.class.getPackage().getImplementationVersion())));
-            });
+        String implementationVersion = JavaFormatExtension.class.getPackage().getImplementationVersion();
+
+        Configuration configuration = rootProject.getConfigurations().create(CONFIGURATION_NAME);
+        configuration.setDescription("Internal configuration for resolving the palantir-java-format implementation");
+        configuration.setVisible(false);
+        configuration.setCanBeConsumed(false);
+        configuration.setCanBeResolved(true);
+        configuration.defaultDependencies(deps -> {
+            deps.add(rootProject
+                    .getDependencies()
+                    .create(ImmutableMap.of(
+                            "group",
+                            "com.palantir.javaformat",
+                            "name",
+                            "palantir-java-format",
+                            "version",
+                            implementationVersion)));
+        });
+
+        Configuration nativeConfiguration = rootProject.getConfigurations().create(NATIVE_CONFIGURATION_NAME);
+        nativeConfiguration.setDescription(
+                "Internal configuration for resolving the palantir-java-format implementation");
+        nativeConfiguration.setVisible(false);
+        nativeConfiguration.setCanBeConsumed(false);
+        nativeConfiguration.setCanBeResolved(true);
+        nativeConfiguration.defaultDependencies(deps -> {
+            deps.add(rootProject
+                    .getDependencies()
+                    .create(ImmutableMap.of(
+                            "group",
+                            "com.palantir.javaformat",
+                            "name",
+                            "palantir-java-format-native",
+                            "version",
+                            implementationVersion,
+                            "classifier",
+                            String.format("nativeImage-%s", getCurrentOs()),
+                            "ext",
+                            "exe")));
```

Comment:
shouldn't have an exe extension for non-windows binaries

Reply:
Unfortunately I need an extension, if I leave the extension empty, the file will be considered as a jar. 
Since on unix I don't think the extension matters, I left it with `exe` here as well - I can change it to something else - but had troubles finding a name for the extension 😅

Reply:
Mmm, I guess we could make it `.bin` for macos/linux. It's not ideal, but I think it's going to confuse any external people using it having it be `.exe`.